### PR TITLE
Bump Ruby to 2.3.7

### PR DESCRIPTION
**Pull request by Matt-Yorkley**
#### What? Why?

Related to #2891 

Bumps Ruby to `2.3.7` :tada: I had to drop down from `2.3.8` because Semaphore apparently doesn't support it. 

Full Ruby changelogs are here:

- 2.3.0 https://svn.ruby-lang.org/repos/ruby/tags/v2_3_0/ChangeLog
- 2.3.1 https://svn.ruby-lang.org/repos/ruby/tags/v2_3_1/ChangeLog
- 2.3.2 https://svn.ruby-lang.org/repos/ruby/tags/v2_3_2/ChangeLog
- 2.3.3 https://svn.ruby-lang.org/repos/ruby/tags/v2_3_3/ChangeLog
- 2.3.4 https://svn.ruby-lang.org/repos/ruby/tags/v2_3_4/ChangeLog
- 2.3.5 https://svn.ruby-lang.org/repos/ruby/tags/v2_3_5/ChangeLog
- 2.3.6 https://svn.ruby-lang.org/repos/ruby/tags/v2_3_6/ChangeLog
- 2.3.7 https://svn.ruby-lang.org/repos/ruby/tags/v2_3_7/ChangeLog

#### What should we test?

General release type test of core functionality.

#### Release notes

Bumped Ruby version to `2.3.7`

Changelog Category: Changed

### CODE REVIEW

**Comment by Matt-Yorkley** [comment]
It's green!

**Comment by luisramos0** [comment]
"Rails 3.2 reached its end of life-cycle with the release of Rails 5.0 on June 30th, 2016"
Ruby 2.3.0  was released on 2015-12-25 
Ruby 2.3.7 was released on 2018-03-28

"Rails versions 3.2 to 4.2 were built to work with any version of Ruby from 1.9. 3 to 2.2, allowing flexibility"

I assume this will be the last ruby upgrade before the rails upgrade. Is that correct?
I think ruby 2.4 will only be ok with Rails 4.2 (not rails 4). [link](https://stackoverflow.com/questions/9087116/which-ruby-on-rails-is-compatible-with-which-ruby-version)
I hope this ruby 2.3 will not create trouble when upgrading to rails 4.0.

**Comment by Matt-Yorkley** [comment]
Yeah I think this will be the last one.

**Comment by mkllnk** [comment]
I'm concerned about Rail 4.0 being untested with Ruby 2.3. I haven't been able to run the Rails specs yet to confirm it it works. I would like test a couple more things. Otherwise I would say let's just try it. Worst case would be that we have to downgrade Ruby during the next upgrade. :-(

File: Gemfile [hunk 1]
```diff
@@ -1,5 +1,5 @@
 source 'https://rubygems.org'
-ruby "2.2.10"
+ruby "2.3.7"
```
Comment (mkllnk) [inline Gemfile]: I'm wondering if we should remove this line altogether. I believe that the purpose is to make Bundler fail fast if the used Ruby version isn't right. [Docs](https://bundler.io/v1.7/gemfile_ruby.html). But our application does work with several Ruby versions at the moment (2.1, 2.2, 2.3). So I would skip this requirement and see the `.ruby-version` file as source of truth to select the best version.

**Comment by sigmundpetersen** [comment]
I guess this is dependent on https://github.com/openfoodfoundation/ofn-install/pull/549 before merge? :blush:

File: Gemfile (hunk 1 above)
Comment (sauloperez) [reply Gemfile]: yes, I think we'll get to it as soon as it breaks our development with the two versions.

**Comment by luisramos0** [comment]
I provisioned staging-fr with latest ofn-install (with 549) and now I started the deploy of this PR:
https://semaphoreci.com/openfoodfoundation/openfoodnetwork-2/servers/fr-staging/deploys/137

**Comment by luisramos0** [comment]
looking good, ready for testing.

**Comment by RachL** [comment]
Ok so I've taken the same testing approach than #4250 and #4301, which means I did a sanity check:

1- Login
2- Create a product and it appears correctly in the shop
3- Checkout (paypal & cash)
4- See the order on the order view (+ capture and ship and edit)

The rest will be tested during release testing like previsouly. So I'm moving this to ready to go :tada:

**Comment by sigmundpetersen** [comment]
@luisramos0 should the open PR's in the pipe be rebased after this merge?

**Comment by sauloperez** [comment]
No need. Now the servers have the current and upcoming Ruby version installed :tada: we learned the lesson.

**Pull request merged**